lib/ash/resource/validation/negate.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Ash.Resource.Validation.Negate do
+  @moduledoc false
+
+  alias Ash.Error.Changes.InvalidAttribute
+
+  @opt_schema [
+    validation: [
+      type: Ash.Resource.Validation.validation_type(),
+      required: true,
+      doc: "The validation module to negate another validation"
+    ]
+  ]
+
+  use Ash.Resource.Validation
+
+  @impl true
+  def init(opts) do
+    with {:ok, opts} <- Spark.OptionsHelpers.validate(opts, @opt_schema),
+         {validation, validation_opts} = opts[:validation],
+         {:ok, _opts} <- validation.init(validation_opts) do
+      {:ok, opts}
+    else
+      {:error, exception} = error when is_binary(exception) ->
+        error
```

Comment:
It ought to be safe to return whatever error that nested init returns. (i.e removing the `else` case here should be okay).